### PR TITLE
flamenco: update fee collection logic

### PR DESCRIPTION
**Pull request by kbhargava-jump**

### CODE REVIEW

**Review by mjain-jump — APPROVED** [review]

File: src/flamenco/runtime/fd_runtime.c [hunk 1]
```diff
@@ -2705,7 +2705,8 @@ void compute_priority_fee(fd_exec_txn_ctx_t const *txn_ctx, ulong *fee, ulong *p
   }
 }
 
-#define ACCOUNT_DATA_COST_PAGE_SIZE ((double)32 * 1024)
+// https://github.com/anza-xyz/agave/blob/2e6ca8c1f62db62c1db7f19c9962d4db43d0d550/sdk/src/fee.rs#L82
+#define ACCOUNT_DATA_COST_PAGE_SIZE fd_ulong_sat_mul(32, 1024)
```
Comment (ibhatt-jumptrading) [inline src/flamenco/runtime/fd_runtime.c:2709]: dont do this. This is better: 32UL * 1024UL

**Review by ibhatt-jumptrading — COMMENTED** [review]
this is not comprehensive. Please fix code style in a follow up

File: src/flamenco/runtime/fd_runtime.c [hunk 2]
```diff
@@ -2760,14 +2738,15 @@ fd_runtime_calculate_fee(fd_exec_txn_ctx_t *txn_ctx,
       num_signatures = fd_ulong_sat_add(num_signatures, (ulong)(data[0]));
     }
   }
-  double signature_fee = (double)fd_runtime_lamports_per_signature(&txn_ctx->slot_ctx->slot_bank) * (double)num_signatures;
+
+  ulong signature_fee = fd_runtime_lamports_per_signature(&txn_ctx->slot_ctx->slot_bank) * num_signatures;
```
[the remark this answers is not in the document]
Comment (ibhatt-jumptrading) [reply src/flamenco/runtime/fd_runtime.c:2742]: should be a sat mul

File: src/flamenco/runtime/fd_runtime.c [hunk 3]
```diff
@@ -2792,27 +2771,29 @@ fd_runtime_calculate_fee(fd_exec_txn_ctx_t *txn_ctx,
   //                    .unwrap_or_default()
   //            });
 
-  double MEMORY_USAGE_COST = ((((double)txn_ctx->loaded_accounts_data_size_limit + (ACCOUNT_DATA_COST_PAGE_SIZE - 1)) / ACCOUNT_DATA_COST_PAGE_SIZE) * (double)FD_VM_HEAP_COST);
-  double loaded_accounts_data_size_cost = FD_FEATURE_ACTIVE(txn_ctx->slot_ctx, include_loaded_accounts_data_size_in_fee_calculation) ? MEMORY_USAGE_COST : 0.0;
-  double total_compute_units = loaded_accounts_data_size_cost + (double)txn_ctx->compute_unit_limit;
+  // https://github.com/anza-xyz/agave/blob/2e6ca8c1f62db62c1db7f19c9962d4db43d0d550/sdk/src/fee.rs#L116
+  ulong MEMORY_USAGE_COST = (((txn_ctx->loaded_accounts_data_size_limit + (ACCOUNT_DATA_COST_PAGE_SIZE - 1)) / ACCOUNT_DATA_COST_PAGE_SIZE) * FD_VM_HEAP_COST);
```
Comment (ibhatt-jumptrading) [inline src/flamenco/runtime/fd_runtime.c:2775]: sat math here please

**Pull request merged**